### ISSUE TITLE
Issue with rtweet

### ISSUE THREAD

**Issue by marcgronwald** [mkearney/rtweet#179]
Dear all,

I have a problem with using rtweet. I'm using R for a very long time and recently learned about rtweet. I installed the package a few weeks ago and everything worked fine. I began building a data-set of tweets; every few days I managed to download about 40,000 tweets using a specific hash tag. Then suddenly an error message like the following appeared:

Error in `$<-.data.frame`(`*tmp*`, app, value = "package_rtweet") : 
  replacement has 1 row, data has 0
In addition: Warning message:
89 - Invalid or expired token.

I have no idea what this means. Downloading worked fine, I did not change anything. Meanwhile I tried to reinstall the package, restarted the R session etc. Also, I revoked access for rtweet in my twitter account. (Initially there was a different warning message; twitter could not authenticate me). The first part of the error message is a problem with the data, the warning message is from twitter. Also tried everything on a different computer. Somehow it seems as if rtweet cannot connect to twitter. This issue is very urgent as I am building this data-set for research purposes and I need a more or less continuous data-set.

Any help more than appreciated!

Marc

**Comment by jcmundy** [comment]
I'm getting a similar error:
```Error in `$<-.data.frame`(`*tmp*`, "app", value = "package_rtweet") : 
  replacement has 1 row, data has 0
In addition: Warning message:
32 - Could not authenticate you.```

I haven't changed anything from last week when it worked.

**Comment by hochoy** [comment]
Same here. I ended up following the vignette below to authenticate via the create_token() method. 
http://rtweet.info/articles/auth.html

**Comment by marcgronwald** [comment]
Thanks for your help; this is incredibly helpful! Now it seems to work again.

**Comment by giannetti** [comment]
Ditto. Thanks so much for sharing your fix, @hochoy!

**Comment by hochoy** [comment]
No worries, I only shared mkearney's vignette. Gotta thank him/her for being detailed in the vignettte documentation. An update must have affected the browser-authentication script but not the underlying API interaction.

**Comment by giannetti** [comment]
Indeed. Thanks to @mkearney for the excellent documentation!

**Comment by mkearney** [comment]
### Update (summary)
Twitter suspended the rtweet app that's embedded in the rtweet package. The app is used to to generate a unique token for users of rtweet, which is the default token-generation method. Based on the messages/notifications I've received, it seems anyone using the default token method is having problems getting Twitter to recognize their requests as authorized. Often this results in an error message about a failure to convert an error message into a data frame (an issue that should be fixed anyway).

As users have pointed out above, the work around is to create your own Twitter app, following instructions contained in the [Obtaining and using access tokens](http://rtweet.info/articles/auth.html) vignette. *Creating your own app/token should immediately resolve any issues you experience related to this issue. This should hold true for all users, assuming user accounts are not otherwise in violation of [Twitter's Developer Policy](https://developer.twitter.com/en/developer-terms/policy)*. 

I'm sorry for not communicating with everyone sooner. I was a bit too optimistic, thinking the issue would be resolved quickly and the package would be back to normal functionality in a matter of hours or [at most] a day or two. The good news is I expect the app to be reinstated any time now. And I don't expect this will occur again in the future.

**tl;dr**: there is a known issue with rtweet's default token-generating process. It will be fixed soon. In the mean time, `rtweet` users are **not** precluded from interacting with Twitter APIs, but some users will have to take a few extra steps to acquire authorization.

### The reason the app was suspended
I use two different apps for maintenance of the package–one for generating unique user tokens and the other for unit testing. Twitter has an [apparently] narrow interpretation of one app per one purpose developer policy. I deleted the unit testing app to comply with their policy immediately; and I've been waiting (at this point several days) for Twitter to reinstate the app. Needless to say, the communication process has been really slow.

### Stability of `rtweet` moving forward
First, assuming the account/app gets reinstated, I don't expect this to ever be a problem again. I have no reason to want to violate Twitter's policies; and for the sake of my email, Github notifications, and the course I teach on collecting and analyzing Twitter data, it's in my best interest to stay in the good graces of Twitter. I'm sorry to anyone who's been affected by this. I've been doing everything I can to get the package back online (for users who haven't created their own app) as soon as possible. 

Second, moving forward, if you're a reliant on Twitter data or a heavy user of rtweet, the most stable path forward is to do what other users have been doing above–create your own Twitter app using the [Obtaining and using access tokens](http://rtweet.info/articles/auth.html) vignette. It's really quite easy to do, and it gives you the added functionality of being able to read DMs and send POST requests (e.g., follow users, post tweets, etc.).

**Comment by 00Mugen00** [comment]
Good day everyone:

Does anybody know if the Twitter issue has been fixed? (Maybe it's my code that is not properly working).
I tried to do the workaround and it works, but it has a small problem. Somehow the token I manually generate (twitter_token.rds according to the vignette) is overwritten by the file generated by the rtweet's default token-generating system (rtweet_token.rds, rtweet_token1.rds if,...). So I was wondering if the Twitter issue been fixed (so that I wont have to manually generate the token) or has someone maybe encountered the same problem as me. If so can you please tell me how you fixed the problem. I would really appreciate your help! and btw nice work @mkearney, this is without a doubt the best library for accessing Twitter API in R.

Have a good day everyone!

**Comment by DevDataAnalyst** [comment]
@mkearney any update on weather your app has been re-instated, I am hoping to submit my University project on this one

**Comment by giannetti** [comment]
Not sure I understand the problem you're having, but the `search_tweets()` function works just fine for me when I use it in this way. To create your Twitter token, follow the vignette mentioned above.

```
## create token named "twitter_token"
twitter_token <- create_token(
  app = appname,
  consumer_key = key,
  consumer_secret = secret)

## gather tweets
rstats <- search_tweets(
  "#rstats", n = 1000, include_rts = FALSE, token = twitter_token
)
```

**Comment by mkearney** [comment]
I submitted a form request yesterday. Hopefully I'll hear back today or tomorrow!

**Comment by DevDataAnalyst** [comment]
Thanks, your work deserves serious appreciation, lets hope we get back in action soon

**Comment by DevDataAnalyst** [comment]
Its working again !!!!

![image](https://user-images.githubusercontent.com/10414546/37189459-8e93bf14-237a-11e8-9f8f-545dc95bc1ab.png)

**Comment by jcmundy** [comment]
Is this working yet?

**Comment by aterhorst** [comment]
http://rtweet.info/articles/auth.html no longer exists.

**Comment by llrs** [comment]
Thanks @aterhorst , the rtweet.info website has been down for a while. It is known and in the next release there isn't any mention to that website.
